stable/rabbitmq-ha/templates/statefulset.yaml

```diff
@@ -46,7 +46,18 @@ spec:
         - name: copy-rabbitmq-config
           image: {{ .Values.busyboxImage.repository}}:{{ .Values.busyboxImage.tag}}
           imagePullPolicy: {{ .Values.busyboxImage.pullPolicy }}
-          command: ['sh', '-c', 'cp /configmap/* /etc/rabbitmq; rm -f /var/lib/rabbitmq/.erlang.cookie']
+          command:
+          - 'sh'
+          - '-c'
+          - 'cp /configmap/* /etc/rabbitmq;'
+          {{- if .Values.rabbitmqErlangCookie }}
+          - "echo \"{{ .Values.rabbitmqErlangCookie }}\" > /var/lib/rabbitmq/.erlang.cookie;"
```

Comment:
I'm not sure what causes your particular issue, but I don't think this change will help. It's the exact same thing the [docker-entrypoint.sh](https://github.com/docker-library/rabbitmq/blob/efdf294e2dde6e77129135b294b75324bf6c7c23/3.7/alpine/docker-entrypoint.sh#L196) does. This change was added in PR https://github.com/helm/charts/pull/4610 and documented in a [comment](https://github.com/helm/charts/pull/4610#issuecomment-391035738) why it was necessary.

I'm okay with your chown command change but the other change will need further proof. You should locally add some debugs to print out the `$RABBITMQ_ERLANG_COOKIE` and `/var/lib/rabbitmq/.erlang.cookie` and figure out what is causing them to not match.

Reply:
Yeah, that was my thinking - if docker-entrypoint.sh was unable (for some reason) to write the cookie, then perhaps kubernetes would be able to, either way for me the code to delete the cookie if it already existed wasn't working, I need to debug more than just the cookie and `RABBITMQ_ERLANG_COOKIE` not matching - that much is clear, what I need to debug is why the slip out of sync - it hasn't happened since I pushed the changes in this PR to our cluster - that doesn't mean it won't happen again, but until it does I can't debug it, do you want me to close the issue, or can I keep it open/on hold until I've got more troubleshooting data to offer?

Reply:
I think you should probably make this PR just have the chmod change and you can open a new PR later if you find a fix for the cookie issue.